crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -665,12 +665,36 @@ where
             Ok(outcome) => {
                 match outcome {
                     CanonicalOutcome::AlreadyCanonical { ref header } => {
-                        debug!(
-                            target: "consensus::engine",
-                            fcu_head_num=?header.number,
-                            current_head_num=?self.blockchain.canonical_tip().number,
-                            "Ignoring beacon update to old head"
-                        );
+                        // On Optimism, the proposers are allowed to reorg their own chain at will.
+                        cfg_if::cfg_if! {
+                            if #[cfg(feautre = "optimism")] {
+                                let optimism = self.chain_spec().optimism;
+                            } else {
+                                let optimism = false;
+                            }
+                        };
+                        if optimism {
+                            debug!(
+                                target: "consensus::engine",
+                                fcu_head_num=?header.number,
+                                current_head_num=?self.blockchain.canonical_tip().number,
+                                "[Optimism] Allowing beacon reorg to old head"
+                            );
```

Comment:
this is still redundant.

we're already using cfg_if, no need to convert cfg_if into a bool.

please move this condition into the cfg_if

Reply:
Addressed in https://github.com/anton-rs/op-reth/pull/147